test/parallel/test-cluster-disconnect-handles.js

```diff
@@ -33,12 +33,25 @@ if (cluster.isMaster) {
   worker.on('message', common.mustCall(message => {
     assert.strictEqual(Array.isArray(message), true);
     assert.strictEqual(message[0], 'listening');
+    let continueRecv = false;
     const address = message[1];
     const host = address.address;
     const debugClient = net.connect({ host, port: common.PORT });
     const protocol = new Protocol();
     debugClient.setEncoding('utf8');
-    debugClient.on('data', data => protocol.execute(data));
+    debugClient.on('data', data => {
+      protocol.execute(data);
+      // It can happen that the first continue was sent before the break event
+      // was received. If that's the case, send also a continue from here so the
+      // worker exits
+      if (data.indexOf('continue') !== -1) {
+        continueRecv = true;
+      } else if ((data.indexOf('break') !== -1) && continueRecv) {
+        const req = protocol.serialize({ command: 'continue' });
+        debugClient.write(req);
+      }
+    });
```

Comment:
Wouldn't it be better to rewrite the protocol.onResponse logic to wait for the break event first before sending the continue command?

Thanks for looking into this, by the way.  Looks like this might fix https://github.com/nodejs/node/issues/3907.